File: HLC01_02Java/src/main/java/org/example/InventoryUtils.java
package org.example;

import java.util.List;
import java.util.Scanner;
import java.util.Iterator;


public class InventoryUtils {
    Scanner scanner = new Scanner(System.in);

    public void addVideogame(List<Videojuego> inventory) {
        System.out.print("Introduce el nombre del videojuego: ");
        String nombre = scanner.nextLine();
        System.out.print("Introduce la plataforma: ");
        String plataforma = scanner.nextLine();
        System.out.print("Introduce la cantidad de horas jugadas: ");
        int horas = Integer.parseInt(scanner.nextLine());

        Videojuego videojuego = new Videojuego(nombre, plataforma, horas);
        inventory.add(videojuego);
    }

    public void deleteVideogame(List<Videojuego> inventory) {
        System.out.print("Introduce el nombre del videojuego que quieres eliminar: ");
        String nombre = scanner.nextLine();
        int counter = 0;

        Iterator<Videojuego> iterator = inventory.iterator();
        while (iterator.hasNext()) {
            Videojuego videojuego = iterator.next();
            if (videojuego.getTitulo().equals(nombre)) {
                iterator.remove(); // Remueve el videojuego utilizando el iterador
                counter++;
            }
        }

        System.out.println(counter + " juegos eliminados");
    }

    public void showInventory(List<Videojuego> inventory) {
        for (Videojuego videojuego : inventory) {
            System.out.println(videojuego.toString());
        }
    }
}

File: HLC01_02Kotlin/src/main/kotlin/InventoryUtils.kt
class InventoryUtils {

    fun addVideogame(inventory: MutableList<Videojuego>) {
        print("Introduce el nombre del videojuego: ")
        val nombre = readln()
        print("Introduce la plataforma: ")
        val plataforma = readln()
        print("Introduce la cantidad de horas jugadas: ")
        val horas = readln().toInt()

        val videojuego = Videojuego(nombre, plataforma, horas)
        inventory.add(videojuego)
    }


    fun deleteVideogame(inventory: MutableList<Videojuego>) {
        print("Introduce el nombre del videojuego que quieres eliminar: ")
        val nombre = readln()
        var counter = 0

        for (videojuego in inventory) {
            if (videojuego.titulo == nombre) {
                inventory.remove(videojuego)
                counter += 1
            }
        }

        println("$counter juegos eliminados")
    }


    fun showInventory(inventory: List<Videojuego>) {
        for (videojuego in inventory) {
            println(videojuego.toString())
        }
    }

}